identity/src/main/java/androidx/security/identity/GattClient.java

```diff
@@ -205,23 +239,88 @@ public void onMtuChanged(@NonNull BluetoothGatt gatt, int mtu, int status) {
     public void onCharacteristicRead(@NonNull BluetoothGatt gatt,
             @NonNull BluetoothGattCharacteristic characteristic,
             int status) {
-        if (!characteristic.getUuid().equals(mCharacteristicIdentUuid)) {
+        if (characteristic.getUuid().equals(mCharacteristicIdentUuid)) {
+            byte[] identValue = characteristic.getValue();
+            if ((mLoggingFlags & Constants.LOGGING_FLAG_TRANSPORT_SPECIFIC) != 0) {
+                Log.d(TAG, "Received identValue: " + Util.toHex(identValue));
+            }
+            // TODO: maybe comment out or change to warning since it's optional... several readers
+            //  send the wrong value (others send the right one though)
+            if (!Arrays.equals(identValue, mIdentValue)) {
+                reportError(new Error("Received ident does not match expected ident"));
+                return;
+            }
+
+            afterIdentObtained(gatt);
+        } else if (characteristic.getUuid().equals(mCharacteristicL2CAPUuid)) {
+            byte[] psmValue = characteristic.getValue();
+            if (psmValue == null || psmValue.length != 1) {
```

Comment:
It appears that the PSM can be up to 4 bytes in length, since the server socket API returns an int.

Additionally, the bluetooth core specification seems to indicate that it's 2 octets minimum.